src/OrchardCore/OrchardCore.DisplayManagement/TagHelpers/ZoneTagHelper.cs

```diff
@@ -41,6 +44,14 @@ public async override Task ProcessAsync(TagHelperContext context, TagHelperOutpu
             {
                 await shape.AddAsync(childContent, Position);
             }
+            else
+            {
+                _logger.LogWarning(
+                    "Unable to add shape to the zone using the <zone> tag helper because the zone's type is " +
+                    "\"{ActualType}\" instead of the expected {ExpectedType}",
+                    zone.GetType().FullName,
+                    nameof(Shape));
```

Comment:
How the `{ExpectedType}` will be different while the `nameof(Shape)` returns "Shape" all the time?!!

Reply:
Not everyone knows about the pre-rendering and if you are developing a theme then it can have logical uses. (In my case I had a shape in the base theme's layout's header that the derived theme can override to inject stuff). 
I didn't know about PositionWrapper earlier and I thought that something else broke. The purpose of this PR is to be more developer friendly by warning about the irregular behavior.